src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -1055,16 +1055,71 @@ public void Connect(IPAddress[] addresses, int port)
                 throw new NotSupportedException(SR.net_invalidversion);
             }
 
+// Disable CS0162: Unreachable code detected
+//
+// SuportsMultipleConnectAttempts is a constant; when false, the following lines will trigger CS0162.
+#pragma warning disable 162
             Exception lastex = null;
-            foreach (IPAddress address in addresses)
+            if (SocketPal.SupportsMultipleConnectAttempts)
```

Comment:
> ...there are others throughout the file for which it would be a lot more cumbersome, so I could understand the desire to keep it like this for consistency with the other uses...

Yes, that's why I've left this code as it is.